x-pack/legacy/plugins/code/server/queue/cancellation_service.test.ts

```diff
@@ -25,11 +25,11 @@ test('Register and cancel cancellation token', async () => {
   const cancelSpy = sinon.spy();
   token.cancel = cancelSpy;
 
-  await service.registerCancelableIndexJob(
-    repoUri,
-    token as CancellationToken,
-    Promise.resolve('resolved')
-  );
+  // make sure the promise won't be fulfilled immediately
+  const promise = new Promise(resolve => {
+    setTimeout(resolve, 100);
+  });
+  await service.registerCancelableIndexJob(repoUri, token as CancellationToken, promise);
```

Comment:
let's add a unit test for the case when the `jobPromise` get rejected`.

Reply:
Added one, similar to the case that the job is normally fulfilled.